### PR TITLE
[bugfix] fix a bug found in v0.7 bug bash in the 'grace' model

### PR DESCRIPTION
**Pull request by hengruizhang98**
## Description
 Rewrite the graph augmentation function. @mufeili

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by hengruizhang98** [comment]
@dgl-bot run g4dn.4xlarge all dmlc/master

File: examples/pytorch/grace/aug.py [hunk 1]
```diff
@@ -5,29 +5,37 @@
...
-    D = x.shape[1]
-    mask_rates = th.FloatTensor(np.ones(D) * drop_prob)
-    masks = th.bernoulli(1 - mask_rates)
+    return ng, feat
 
+def drop_feature(x, drop_prob):
+    drop_mask = th.empty((x.size(1),),
+                        dtype=th.float32,
+                        device=x.device).uniform_(0, 1) < drop_prob
     x = x.clone()
```
Comment (mufeili) [inline examples/pytorch/grace/aug.py:30]: Why do we need `x.clone()`?

File: examples/pytorch/grace/aug.py [hunk 2]
```diff
@@ -5,29 +5,37 @@
...
-    x[:, masks] = 0
+    x[:, drop_mask] = 0
+
+    return x
 
-    return x
\ No newline at end of file
+def mask_edge(graph, mask_prob):
+    E = graph.number_of_edges()
```
Comment (mufeili) [inline examples/pytorch/grace/aug.py:36]: `number_of_edges` -> `num_edges`

File: examples/pytorch/grace/aug.py [hunk 3]
```diff
@@ -5,29 +5,37 @@
 import dgl
 
 def aug(graph, x, feat_drop_rate, edge_mask_rate):
-    ng = drop_edge(graph, edge_mask_rate)
-    feat = drop_feat(x, feat_drop_rate)
-    ng = ng.add_self_loop()
+    n_node = graph.number_of_nodes()
```
Comment (mufeili) [inline examples/pytorch/grace/aug.py:8]: `number_of_nodes` -> `num_nodes`

File: examples/pytorch/grace/aug.py [hunk 4]
```diff
@@ -5,29 +5,37 @@
...
+    ng.add_nodes(n_node)
+    src = graph.edges()[0]
+    dst = graph.edges()[1]
 
-    mask_rates = th.FloatTensor(np.ones(E) * drop_prob)
-    masks = th.bernoulli(1 - mask_rates)
-    edge_idx = masks.nonzero().squeeze(1)
+    nsrc = src[edge_mask]
+    ndst = dst[edge_mask]
+    ng.add_edges(nsrc, ndst)
```
Comment (mufeili) [inline examples/pytorch/grace/aug.py:20]: You can directly do `ng = dgl.graph((nsrc, ndst), num_nodes=n_node)`

File: examples/pytorch/grace/aug.py (hunk 1 above)
Comment (hengruizhang98) [reply examples/pytorch/grace/aug.py:30]: Otherwise the input feature mask would be changed.

Comment (mufeili) [reply examples/pytorch/grace/aug.py:28]: Ok

**Review by mufeili — APPROVED** [review]
LGTM

**Pull request merged**